libraries/Microsoft.Bot.Builder/TurnContextServiceCollection.cs

```diff
@@ -46,95 +22,42 @@ public TurnContextServiceCollection()
         {
             if (key == null) throw new ArgumentNullException(nameof(key));
 
-            if(!_services.TryGetValue(key, out var service))
+            if(TryGetValue(key, out var service))
             {
-                // TODO: log that we didn't find the requested service
+                if (service is TService result)
+                {
+                    return result;
+                }
```

Comment:
else
{
    throw new InvalidCastException("adasdfasdf")
}

Reply:
OK I can do that... I think your right about the semantics - however note the semantics in the original code were very sketchy - basically null

I should also add a similar TryGetValue<Type> method that doesn't throw?

Reply:
changing the semantics to the return NULL it was originally.